probe/endpoint/procspy/background_reader_linux.go

```diff
@@ -0,0 +1,144 @@
+package procspy
+
+import (
+	"bytes"
+	"io"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/weaveworks/scope/probe/process"
+)
+
+const (
+	initialRateLimitPeriod = 50 * time.Millisecond  // Read 20 * fdBlockSize file descriptors (/proc/PID/fd/*) per namespace per second
+	maxRateLimitPeriod     = 500 * time.Millisecond // Read at least 2 * fdBlockSize file descriptors per namespace per second
+	minRateLimitPeriod     = initialRateLimitPeriod
+	fdBlockSize            = uint64(300) // Maximum number of /proc/PID/fd/* files to stat per rate-limit period
+	// (as a rule of thumb going through each block should be more expensive than reading /proc/PID/tcp{,6})
+	targetWalkTime = 10 * time.Second // Aim at walking all files in 10 seconds
+)
+
+type backgroundReader struct {
+	stopc         chan struct{}
+	mtx           sync.Mutex
+	latestBuf     *bytes.Buffer
+	latestSockets map[uint64]*Proc
```

Comment:
I don't think it is, but that's the format it had, (and what's expected in the output) and I don't know if it will have performance implications. I would like to avoid refactoring `procspy` more than I have in this PR.